pkg/provider/azure_vmss_cache.go

```diff
@@ -484,22 +484,47 @@ func (ss *ScaleSet) getVMManagementTypeByIPConfigurationID(ipConfigurationID str
 		return ManagedByUnknownVMSet, err
 	}
 
-	matches := nicIDRE.FindStringSubmatch(ipConfigurationID)
-	if len(matches) != 3 {
+	nicResourceGroup, nicName, err := getResourceGroupAndNameFromNICID(ipConfigurationID)
```

Comment:
Could you add some comments on how the VM type is got for function getVMManagementTypeByIPConfigurationID?

Reply:
added